src/Draco.Compiler/Semantics/Symbol.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Generic;
+
+namespace Draco.Compiler.Semantics;
+
+/// <summary>
+/// Defines semantic symbols.
+/// </summary>
+public static class Symbol
+{
+    /// <summary>
+    /// A base symbol.
+    /// </summary>
+    public interface ISymbol
+    {
+        /// <summary>
+        /// The name of the symbol.
+        /// </summary>
+        string Name { get; }
+    }
+
+    /// <summary>
+    /// A variable symbol.
+    /// </summary>
+    public interface IVariable : ISymbol
+    {
+        /// <summary>
+        /// The type of the variable.
+        /// </summary>
+        IType Type { get; }
+        /// <summary>
```

Comment:
Please add a newline between these. If you know of the editorconfig rule, that would also be handly.